config/areas/users.php

```diff
@@ -0,0 +1,91 @@
+                            'id'    => $role->id(),
+                            'title' => $role->title(),
+                        ];
+                    });
+
+                    return [
+                        'component' => 'k-users-view',
+                        'props'     => [
+                            'role' => function () use ($kirby, $roles, $role) {
+                                if ($role) {
+                                    return $roles[$role] ?? null;
+                                }
+                            },
+                            'roles' => array_values($roles),
+                            'users' => function () use ($kirby, $role) {
+                                $users = $kirby->users();
+
+                                if (empty($role) === false) {
+                                    $users = $users->role($role);
+                                }
+
+                                $users = $users->paginate([
+                                    'limit' => 20,
+                                    'page'  => get('page')
+                                ]);
+
+                                return [
+                                    'data' => $users->values(function ($user) {
+                                        return [
+                                            'id'   => $user->id(),
+                                            'icon' => [
+                                                'type' => 'user',
+                                                'back' => 'black'
+                                            ],
+                                            'text'  => $user->username(),
+                                            'info'  => $user->role()->title(),
+                                            'link'  => 'users/' . $user->id(),
+                                            'image' => true
+                                        ];
+                                    }),
+                                    'pagination' => $users->pagination()->toArray()
+                                ];
+                            },
+                        ]
+                    ];
+                }
+            ],
+            [
+                'pattern' => 'users/(:any)',
+                'action'  => function (string $id) use ($kirby) {
+                    if (!$user = $kirby->user($id)) {
+                        return t('error.user.undefined');
+                    }
+
+                    return $user->panel()->route();
+                }
+            ],
+            [
+                'pattern' => 'users/(:any)/files/(:any)',
+                'action'  => function (string $id, string $filename) use ($kirby) {
+                    $filename = urldecode($filename);
+
+                    if (!$user = $kirby->user($id)) {
+                        return t('error.user.undefined');
+                    }
+
+                    if (!$file = $user->file($filename)) {
```

Comment:
```suggestion
                    $file = $user->file($filename);
                    if (!$file) {
```

Reply:
I really love the onliner. Why would you change it?

Reply:
Setting variables in `if` blocks is [discouraged](https://phpmd.org/rules/cleancode.html#ifstatementassignment) because it can cause unexpected issues that are hard to spot. So far we mostly use it like `if ($var = $object->method())` (which is fine in the cases we use it in IMO) but with the additional `!` thrown into the mix this gets quite complex and I fear that it might break in future PHP releases. In any case this is quite hard to parse mentally.